src/main/java/TreeHandlers/TreeHandler.java

```diff
@@ -0,0 +1,80 @@
+package TreeHandlers;
+
+import Entities.*;
+import Interactors.GameLogic;
+
+import java.util.HashMap;
+
+public class TreeHandler {
+    GameLogic gameLogicInteractor;
+    Player currentPlayer;
+    Board board;
+    HashMap<String, Integer> selectedOptions = new HashMap<String, Integer>();
+    int returnPlayerIndex = -1;
+    Player[] players;
+
+    public void initialize(Player currentPlayer, Board board){
+        players = board.getPlayers();
+        this.currentPlayer = currentPlayer;
+        this.board = board;
+    }
+    public void setGameLogicInteractor(GameLogic interactor){
+        gameLogicInteractor = interactor;
+    }
+
+    public State getInitialState(){
+        State currentState = new State();
+        for (MenuTree tree: gameLogicInteractor.currentTree.getChildren()){
+            currentState.addOptions(tree.getName());
+        }
+        return currentState;
+    }
+    public int getCurrentPlayerIndex(){
+        for (int i = 0; i< players.length; i++){
+            if (currentPlayer == players[i]){
+                return i;
+            }
+        }
+        throw new RuntimeException("Player not in array");
+    }
+    public Object[][] playersToArray(){
```

Comment:
please remember to add JavaDoc in the future so it is easy to understand

Reply:
Yes, JavaDoc will also be added soon